src/mmw/js/src/modeling/tests.js

```diff
@@ -556,21 +693,92 @@ describe('Modeling', function() {
-                    assert.isTrue(saveSpy.called, 'attemptSave should have been called');
-                    assert.isTrue(resultSpy.calledBefore(saveSpy));
+            describe('#getResults', function() {
+                beforeEach(function() {
+                    this.results = new models.ResultCollection();
+                    this.setPollingSpy = sinon.spy(this.results, 'setPolling');
+                    this.setNullResultsSpy = sinon.spy(this.results, 'setNullResults');
+                    this.scenarioModel = new models.ScenarioModel({});
+                    this.setResultsSpy = new sinon.spy(this.scenarioModel, 'setResults');
+                    this.scenarioModel.set('results', this.results);
+                    this.server = sinon.fakeServer.create();
+                    this.server.respondImmediately = true;
+                    this.startJob = '1';
+                    this.startResponse = {
+                        job: this.startJob
+                    };
+                    this.server.respondWith('POST', '/api/modeling/start/tr55/',
+                                            [ 200,
+                                              { 'Content-Type': 'application/json' },
+                                              JSON.stringify(this.startResponse) ]);
+                    this.pollingResponse = mocks.polling.getJobSuccess;
+                    this.server.respondWith('GET', '/api/modeling/jobs/1/',
+                                            [ 200,
+                                              { 'Content-Type': 'application/json' },
+                                              this.pollingResponse ]);
+                });
+
+                function getResultsAssertions(self) {
+                    assert(inputModSpy.calledTwice, 'updateInputModHash should have been called twice');
+                    assert(self.setPollingSpy.calledTwice, 'setPolling should have been called twice');
+                    assert(self.setPollingSpy.getCall(0).args[0], 'should have initially called setPolling(true)');
+                    assert.isFalse(self.setPollingSpy.getCall(1).args[0], 'should have subsequently called setPolling(false)');
+                }
+
+                it('sets results to null on start failure', function(done) {
+                    var self = this;
+                    self.server.respondWith('POST', '/api/modeling/start/tr55/',
+                                            [ 400, // Make request fail.
+                                              { 'Content-Type': 'application/json' },
+                                              JSON.stringify(self.startResponse) ]);
+
+                    self.scenarioModel.getResults().startPromise.always(function() {
+                        assert(self.setNullResultsSpy.calledOnce, 'setNullResults should have been called once');
+                        assert(self.setResultsSpy.callCount === 0, 'setResults should not have been called');
```

Comment:
I knew there had to be a better way. Taking it a step further, I can do `assert.isFalse(self.setResultsSpy.called)`.